diff --git a/make_helpers/defaults.mk b/make_helpers/defaults.mk
index 8829fcb0e..e444d83ec 100644
--- a/make_helpers/defaults.mk
+++ b/make_helpers/defaults.mk
@@ -330,13 +330,6 @@ COT_DESC_IN_DTB			:= 0
 # Build option to provide OpenSSL directory path
 OPENSSL_DIR			:= /usr
 
-# Select the openssl binary provided in OPENSSL_DIR variable
-ifeq ("$(wildcard ${OPENSSL_DIR}/bin)", "")
-    OPENSSL_BIN_PATH = ${OPENSSL_DIR}/apps
-else
-    OPENSSL_BIN_PATH = ${OPENSSL_DIR}/bin
-endif
-
 # Build option to use the SP804 timer instead of the generic one
 USE_SP804_TIMER			:= 0
 
